### PR TITLE
Comm archive

### PR DESCRIPTION
**Pull request by sjanssen2**
Adding a new (currently not used) parameter for the reference tree/alignment for SEPP placements.
Actually talk to archive to retrieve pre-computed placements, compute placements via SEPP for novel fragments and store novel placements in archive.

### CODE REVIEW

**Comment by sjanssen2** [comment]
This PR will fail as long as https://github.com/biocore/qiita/pull/2460 is not merged.

File: qp_deblur/__init__.py [hunk 1]
```diff
@@ -42,7 +42,9 @@
     'Minimum dataset-wide read threshold': ['integer', '0'],
     'Minimum per-sample read threshold': ['integer', '2'],
     'Threads per sample': ['integer', '1'],
-    'Jobs to start': ['integer', '1']
+    'Jobs to start': ['integer', '1'],
+    'Reference phylogeny for SEPP': ['choice:["Greengenes_13.8"]',
```
Comment (antgonza) [inline qp_deblur/__init__.py]: This should be the path to the actual tree, perhaps worth updating?

Comment (antgonza) [reply qp_deblur/__init__.py]: Looking at the code below, we could have for example: /bla/ble/bli/file and the code below can use .nwk/fasta. Thoughts?

Comment (sjanssen2) [reply qp_deblur/__init__.py]: I was thinking about the name that is facing the user in the end and I figured "Greengenes_13.8" is more speaking than /bla/ble/bli/gg13.8/. Therefore, my code below makes an explicit mapping of the "name" to the actual files or the default files.

Easy to change, but we need to decide what the user shall see.

**Comment by sjanssen2** [comment]
ready for another round of reviews @antgonza

File: qp_deblur/deblur.py [hunk 2]
```diff
@@ -415,19 +420,84 @@ def deblur(qclient, job_id, parameters, out_dir):
...
                                                      'features': features})
-        no_placements = [k for k, v in observations.items() if v == '']
-        qclient.update_job_step(job_id, "Step 4 of 4 (2/2): Generating %d new "
-                                "placements" % len(no_placements))
+        novel_fragments = list(set(features) - set(observations.keys()))
+
+        qclient.update_job_step(job_id, "Step 4 of 4 (2/4): Generating %d new "
+                                "placements" % len(novel_fragments))
         try:
+            # Once we support alternative reference phylogenies for SEPP in the
```
Comment (antgonza) [inline qp_deblur/deblur.py]: This is a pretty large try block, can we shorten? The issue with long blocks like this is that we are not sure why/where they raise the issue.

**Comment by sjanssen2** [comment]
fixed coverage and decreased try blocks. What do you think @antgonza ?

**Comment by sjanssen2** [comment]
Thank you!

**Pull request merged**